rds/rds_provider.go

```diff
@@ -30,28 +30,41 @@ type RDS struct {
 func New(db *crd.Database, kc *kubernetes.Clientset) (*RDS, error) {
 	ec2client, err := ec2client(kc)
 	if err != nil {
-		log.Fatal("unable to create a client for EC2 ", err)
+		log.Fatal("Unable to create a client for EC2 ", err)
 	}
 
 	nodeInfo, err := describeNodeEC2Instance(kc, ec2client)
 	if err != nil {
 		log.Println(err)
-		return nil, errors.Wrap(err, "unable AWS metadata")
+		return nil, errors.Wrap(err, "Unable AWS metadata")
 	}
 
 	vpcId := *nodeInfo.Reservations[0].Instances[0].VpcId
 
-	log.Println("trying to get subnets")
-	subnets, err := getSubnets(nodeInfo, ec2client, db.Spec.PubliclyAccessible)
-	if err != nil {
-		return nil, fmt.Errorf("unable to get subnets from instance: %v", err)
+	log.Println("Trying to get subnets")
+	subnets := db.Spec.Subnets
+	if db.Spec.DBSubnetGroupName == "" || len(subnets) < 2 {
```

Comment:
- Re validation: the property just can't be 1 any other value is ok, I can add some extra validation logic in the CRD?
- Re log: I put the log statement in the get subnets function since the information for the node id is more readily available making the log statement more detailed.